### PR TITLE
Add an API for setting settings

### PR DESCRIPTION
**Pull request by filiptronicek**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -38,6 +38,9 @@ model User {
   name          String?
   emailVerified DateTime?
   image         String?
+  storageProvider String? @default("s3")
```
Comment (bigint) [inline prisma/schema.prisma:41]: I think its typo, its meant to be ipfs rit? 🤣

File: src/components/Settings/SettingsCard.tsx [hunk 2]
```diff
@@ -25,9 +25,11 @@ const SettingsCard = ({
       <div className="p-4">{children}</div>
       <div className="flex w-full p-2 border-t border-gray-300 space-between bg-[#FAFBFB] dark:bg-[#2f2f2f] h-14 rounded-b-xl dark:border-gray-700">
         <p className="w-full px-4 my-auto">{footerDescription}</p>
-        <Button color={warning ? 'error' : 'primary'} auto onClick={onSave}>
-          {buttonText ?? 'Save'}
-        </Button>
+        {onSave && (
+          <Button color={warning ? 'error' : 'primary'} auto onClick={onSave}>
```
Comment (bigint) [inline src/components/Settings/SettingsCard.tsx:29]: color is not a correct attribute name I think, IMO variant sounds good

File: src/pages/api/account/setDetails.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import { db } from '@utils/prisma';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getSession } from 'next-auth/react';
+
+/**
+ * Changes user settings
+ * @param setProperties an array of key value pairs of user fields to be updated. Their values need to be strings for now, but maybe in the futrure we can transfer them via JSON
```
Comment (bigint) [inline src/pages/api/account/setDetails.ts:7]:
```suggestion
 * @param setProperties an array of key-value pairs of user fields to be updated. Their values need to be strings for now, but maybe in the future we can transfer them via JSON
```

File: src/pages/api/account/setDetails.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ * @param req the HTTP request
+ */
+export const setUserDetails = async (
+  setProperties: { [key: string]: string },
+  req: NextApiRequest,
+) => {
+  const session = await getSession({ req });
+
+  if (!session?.user?.email) {
+    // eslint-disable-next-line prettier/prettier
```
Comment (bigint) [inline src/pages/api/account/setDetails.ts:17]: 🤔

**Review by bigint — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma (hunk 1 above)
Comment (filiptronicek) [reply prisma/schema.prisma:41]: One day, but now to be serious it's better for the common folks to use S3

File: src/components/Settings/SettingsCard.tsx (hunk 2 above)
Comment (filiptronicek) [reply src/components/Settings/SettingsCard.tsx:29]: Can't change that it's built-in to Next-ui :/

File: src/pages/api/account/setDetails.ts (hunk 4 above)
Comment (filiptronicek) [reply src/pages/api/account/setDetails.ts:17]: I have no idea how to fix this, prettier and eslint both seem to have different rules for using `''` when there is a `'` in the string

Comment (filiptronicek) [reply src/pages/api/account/setDetails.ts]: ok solved 😄

**Pull request merged**